dbms/src/Interpreters/Aggregator.cpp

```diff
@@ -1200,6 +1257,29 @@ void NO_INLINE Aggregator::convertToBlockImplFinal(
     });
 }
 
+template <typename Method, typename Table>
+void NO_INLINE Aggregator::convertToBlocksImplFinal(
+    Method & method,
+    Table & data,
+    std::vector<std::vector<IColumn *>> key_columns_vec,
+    std::vector<MutableColumns> & final_aggregate_columns_vec,
+    Arena * arena) const
+{
+    assert(!key_columns_vec.empty());
+    auto shuffled_key_sizes = method.shuffleKeyColumns(key_columns_vec[0], key_sizes);
+    const auto & key_sizes_ref = shuffled_key_sizes ? *shuffled_key_sizes : key_sizes;
+
+    AggregatorMethodInitKeyColumnHelper<Method> agg_keys_helper{method};
+    agg_keys_helper.initAggKeys(data.size(), key_columns_vec[0]);
+
+    size_t index = 0;
+    data.forEachValue([&](const auto & key, auto & mapped) {
+        agg_keys_helper.insertKeyIntoColumns(key, key_columns_vec[index / params.max_block_size], key_sizes_ref, params.collators);
```

Comment:
use local variable to record index / params.max_block_size.

Reply:
> use local variable to record index / params.max_block_size.

done.